include/deal.II/dofs/dof_accessor.templates.h

```diff
@@ -1588,6 +1588,223 @@ namespace internal
+        // orientation, we look at it in flipped orientation and we will have to
+        // adjust the shape function indices that we see to correspond to the
+        // correct (cell-local) ordering. The same applies, if the face_rotation
+        // or face_orientation is non-standard
+        if (structdim == 3)
+          for (unsigned int quad = 0;
+               quad < GeometryInfo<structdim>::quads_per_cell;
+               ++quad)
+            for (unsigned int d = 0; d < dofs_per_quad; ++d, ++index)
+              Operation::process_dof(
+                *accessor.quad(quad),
+                accessor.get_fe(fe_index)
+                  .adjust_quad_dof_index_for_face_orientation(
+                    d,
+                    accessor.face_orientation(quad),
+                    accessor.face_flip(quad),
+                    accessor.face_rotation(quad)),
+                dof_indices[index],
+                fe_index);
+
+        // 4) INNER dofs
+        for (unsigned int d = 0; d < inner_dofs; ++d, ++index)
+          Operation::process_dof(accessor, d, dof_indices[index], fe_index);
+
+        AssertDimension(dof_indices.size(), index);
+      }
+
+
+
+      template <typename DoFHandlerType, bool level_dof_access, int structdim>
+      class DoFIndexGetter
+      {
+      public:
+        static DEAL_II_ALWAYS_INLINE void
+        process_vertex_dof(
+          const dealii::DoFAccessor<structdim, DoFHandlerType, level_dof_access>
+            &                      accessor,
+          const unsigned int       vertex,
+          const unsigned int       d,
+          types::global_dof_index &index_value,
+          const unsigned int       fe_index)
+        {
+          index_value = accessor.vertex_dof_index(vertex, d, fe_index);
+        }
+
+        template <int structdim_>
+        static DEAL_II_ALWAYS_INLINE void
+        process_dof(const dealii::DoFAccessor<structdim_,
+                                              DoFHandlerType,
+                                              level_dof_access> &accessor,
+                    const unsigned int                           d,
+                    types::global_dof_index &                    index_value,
+                    const unsigned int                           fe_index)
+        {
+          index_value = accessor.dof_index(d, fe_index);
+        }
+
+        template <int structdim_>
+        static DEAL_II_ALWAYS_INLINE void
+        process_dof(
+          const dealii::DoFInvalidAccessor<structdim_,
+                                           DoFHandlerType::dimension,
+                                           DoFHandlerType::space_dimension> &,
+          const unsigned int,
+          types::global_dof_index &,
+          const unsigned int)
+        {
+          Assert(false, ExcInternalError());
+        }
+      };
+
+
+
+      template <typename DoFHandlerType, bool level_dof_access, int structdim>
+      class DoFIndexSetter
```

Comment:
Even though it is internal, I would still suggest to add a comment for this struct - and let it be a struct rather than a class because there is no access restriction.